src/ctl/ctl.c

```diff
@@ -324,13 +355,17 @@ int ctl_query(struct ctl *ctl, void *ctx, enum ctl_query_source source,
         n = ctl_find_node(ctl->root, name, indexes);
     }
 
-    if (n == NULL || n->type != CTL_NODE_LEAF || n->cb[type] == NULL) {
+    if (n == NULL ||
+        (n->type != CTL_NODE_LEAF && n->type != CTL_NODE_SUBTREE) ||
+        n->cb[n->type == CTL_NODE_SUBTREE ? CTL_QUERY_SUBTREE : type] == NULL) {
         errno = EINVAL;
         goto out;
     }
 
-    ret = ctl_exec_query[type](ctx, n, source, arg, indexes);
-
+    char *extra_name = strstr(name, n->name);
```

Comment:
This do not work correctly. First of all strstr is not "safe" 
But also this finds "by_handle" node. So you pass to provider string which beggins from "by_handle" string. In find_node function we already parse ctl string, so you should get the information about when the name string is split, inside of this function

Reply:
Done